fix dragonAIMate bug and one NPE crash

ClaymanTwinkle opened the pull request:
1. fix dragonAIMate bug
find the nearest dragon which is in love 
2. fix NPE crash
3. add dragon part name

ClaymanTwinkle committed:
fix dragon ai mate bug

ClaymanTwinkle committed:
revert

ClaymanTwinkle committed:
dragon part name

ClaymanTwinkle committed:
revert

ClaymanTwinkle committed:
fix warning

ClaymanTwinkle committed:
add dragon armor effect;

ClaymanTwinkle committed:
fix silver bug;
fix seaserpentJumping1.tbl not found bug;

ClaymanTwinkle committed:
fix ai attack melee bug;

ClaymanTwinkle committed:
fix warnning

ClaymanTwinkle committed:
fix dragon ai mate;
fix warnning

ClaymanTwinkle committed:
fix mate task is not excuted

ClaymanTwinkle committed:
fix server thread delay bug

ClaymanTwinkle committed:
dragon don't attack it's owner

ClaymanTwinkle committed:
fix dragon dismiss when using sword attack dragon

ClaymanTwinkle committed:
fix dragon dismiss when using sword attack dragon

ghost commented:
This fixes #1575 
Works fine here.
Alex will need to review this before merging it to master. This fixes a fair few issues and can be safely used with the latest build of ice and fire.
I don’t know why this wasn’t merged two months ago...

ClaymanTwinkle committed:
fix crash when using splash

ClaymanTwinkle committed:
fix dragon run away;
fix when owner be attacked and the dragon is sleeping, dragon do not help owner;

ClaymanTwinkle committed:
fix dragon run away;

ClaymanTwinkle committed:
clean code

ClaymanTwinkle committed:
clean code and fix bug

ClaymanTwinkle committed:
try to fix fly

ClaymanTwinkle committed:
fix crash

ClaymanTwinkle committed:
clear code

ClaymanTwinkle committed:
clear code

ClaymanTwinkle committed:
fix dragon always in sky

ClaymanTwinkle committed:
clean code

ClaymanTwinkle committed:
clear code

ClaymanTwinkle committed:
fix error

ClaymanTwinkle committed:
fix dragon can hit dragon which is the same owner

ClaymanTwinkle committed:
fix dragon projectile hit owner pet bug

ClaymanTwinkle committed:
fix dragon shoot fire when it dead bug;
clean code

ClaymanTwinkle committed:
save temp

ClaymanTwinkle committed:
fix dragon can not eat food item;
set dragon home when it born;

ClaymanTwinkle committed:
clean code

ClaymanTwinkle committed:
make the tamed dragon attack mob

ClaymanTwinkle committed:
clean code

ClaymanTwinkle committed:
clean code
